colossalai/utils/profiler/common.py

```diff
@@ -0,0 +1,83 @@
+from pathlib import Path
+from typing import List, Union
+
+from colossalai.core import global_context as gpc
+
+from ._base_profiler import BaseProfiler
+
+
+class ProfilerContext(object):
+    '''
+    Profiler context manager
+
+    Usage:
+    from colossalai.utils.profiler import ProfilerActivity, profiler
+    from torch.utils.tensorboard import SummaryWriter
+
+    with profiler(activities=[ProfilerActivity.COMMUNICATION]) as prof:
+        train()
+    writer = SummaryWriter('tb/path')
+    prof.to_tensorboard(writer)
+    prof.to_file('./prof_logs/')
+    prof.show()
+    '''
+
+    def __init__(self, activities: List[BaseProfiler] = None):

```

Comment:
shall we change the parameter name as the type is a profiler

Reply:
I think 'activities' also makes sense, and maybe easier for pytorch users to distinguish the specific profilers from the profiler context (just too many 'profiler's).

Reply:
@FrankLeeeee BTW, ready to merge if there is no other problems.